runtime/bindings/python/iree/runtime/io.py

```diff
@@ -64,3 +67,137 @@ def save_archive_file(entries: dict[str, Union[Any, SplatValue]], file_path: Pat
         else:
             index.add_buffer(key, value)
     index.create_archive_file(str(file_path))
+
+
+def parameter_index_add_numpy_ndarray(
+    index: ParameterIndex, name: str, array: np.ndarray
+):
+    """Adds an named array to the index."""
```

Comment:
two points here:

* The "nd" in "ndarray" stands for "N-dimensional", not "named": https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html
* grammar nit:  "a named" since "named" does not start with a vowel sound... but then perhaps confusingly "an ndarray" is correct (as shown on that page too), since "ndarray" when spoken _does_ start with a vowel sound "en dee array"

```suggestion
    """Adds an ndarray to the index."""
```